core/formats/columnstore2.cpp

```
@@ -1332,7 +1336,13 @@ void column::reset() {
 
   [[maybe_unused]] const bool res = docs_writer_.erase(pend_);
   IRS_ASSERT(res);
-  data_.stream.truncate(addr_table_.back());
+
+  //  TODO:
+  //  back()/push_back() can fail. Inspect the return
+  //  values and handle failures.
+  uint64_t lastElem;
+  addr_table_.back(lastElem);
+  data_.stream.truncate(lastElem);
```

Why not just return value from the function? The boolean value which you return is unused here

---

The return value is unused for now because the implementation presumed that we'll never need more than 64K blocks. I've left a TODO there to fix that and to remove the swallowing of any potential errors.

---

Agree with @k0ushal